### PR TITLE
QSP - Code Coverage

### PR DESCRIPTION
**Pull request by ryley-o**
Add tests to reach 100% statement coverage and >=90% branch coverage for V3 and related minter-suite contracts.

This update brings us to the following code coverages, to meet the QSP recommendation of at least 90% branch-coverage on all minters.

| File | % Stmts | % Branch | % Lines
| --- | --- | --- | --- |
| GenArt721CoreV3.sol | 100 | 100 | 100 |
| MinterFilterV1.sol | 100 | 100 | 100 |
| MinterDAExpV2.sol | 100 | 100 | 100 |
| MinterDALinV2.sol | 100 | 98 | 100 |
| MinterHolderV1.sol | 100 | 97.62 | 100 |
| MinterMerkleV1.sol | 100 | 100 | 100 |
| MinterSetPriceV2.sol | 100 | 100 | 100 |
| MinterSetPriceERC20V2.sol | 100 | 98 | 100 |

## Audit
Please only evaluate the code coverage of the audited minter contracts.

### CODE REVIEW

File: test/minter-suite-minters/SetPrice/MinterSetPriceV2.common.ts [hunk 1]
```diff
@@ -24,6 +24,104 @@ export const MinterSetPriceV2_Common = async () => {
...
+          .connect(this.accounts.deployer)
+          .updateArtblocksPrimarySalesPercentage(0);
+        // update artist primary split to zero
+        const proposedAddressesAndSplits = [
+          this.projectZero,
+          this.accounts.artist.address,
+          this.accounts.additional.address,
+          100,
+          this.accounts.additional2.address,
+          50,
```
Comment (jakerockland) [inline test/minter-suite-minters/SetPrice/MinterSetPriceV2.common.ts:107]: Nit: these two constant values could be helpful to add smol dev comments for.

File: test/minter-suite-minters/SetPrice/MinterSetPriceV2.common.ts [hunk 2]
```diff
@@ -24,6 +24,104 @@ export const MinterSetPriceV2_Common = async () => {
...
+
+      it("requires successful payment to artist additional payee", async function () {
+        // update artist additional payee to a contract that reverts on receive
+        const proposedAddressesAndSplits = [
+          this.projectZero,
+          this.accounts.artist.address,
+          this.deadReceiver.address,
+          50,
+          this.accounts.additional2.address,
+          50,
```
Comment (jakerockland) [inline test/minter-suite-minters/SetPrice/MinterSetPriceV2.common.ts:76]: Nit: these two constant values could be helpful to add smol dev comments for.

File: test/minter-suite-minters/SetPrice/MinterSetPriceERC20/MinterSetPriceERC20V2.test.ts [hunk 3]
```diff
@@ -132,6 +132,77 @@ describe("MinterSetPriceERC20V2_V3Core", async function () {
...
+        .connect(this.accounts.deployer)
+        .updateArtblocksPrimarySalesPercentage(0);
+      // update artist primary split to zero
+      const proposedAddressesAndSplits = [
+        this.projectZero,
+        this.accounts.artist.address,
+        this.accounts.additional.address,
+        100,
+        this.accounts.additional2.address,
+        50,
```
Comment (jakerockland) [inline test/minter-suite-minters/SetPrice/MinterSetPriceERC20/MinterSetPriceERC20V2.test.ts:193]: Nit: these two constant values could be helpful to add smol dev comments for.

File: test/minter-suite-minters/MinterMerkle/MinterMerkleV1.test.ts [hunk 4]
```diff
@@ -197,6 +211,127 @@ describe("MinterMerkleV1", async function () {
...
+        .connect(this.accounts.deployer)
+        .updateArtblocksPrimarySalesPercentage(0);
+      // update artist primary split to zero
+      const proposedAddressesAndSplits = [
+        this.projectZero,
+        this.accounts.artist.address,
+        this.accounts.additional.address,
+        100,
+        this.accounts.additional2.address,
+        50,
```
Comment (jakerockland) [inline test/minter-suite-minters/MinterMerkle/MinterMerkleV1.test.ts:314]: Nit: these two constant values could be helpful to add smol dev comments for.

File: test/minter-suite-minters/MinterMerkle/MinterMerkleV1.test.ts [hunk 5]
```diff
@@ -197,6 +211,127 @@ describe("MinterMerkleV1", async function () {
...
+
+    it("requires successful payment to artist additional payee", async function () {
+      // update artist additional payee to a contract that reverts on receive
+      const proposedAddressesAndSplits = [
+        this.projectZero,
+        this.accounts.artist.address,
+        this.deadReceiver.address,
+        50,
+        this.accounts.additional2.address,
+        50,
```
Comment (jakerockland) [inline test/minter-suite-minters/MinterMerkle/MinterMerkleV1.test.ts:279]: Nit: these two constant values could be helpful to add smol dev comments for.

File: test/minter-suite-minters/MinterHolder/MinterHolderV1.test.ts [hunk 6]
```diff
@@ -185,6 +185,133 @@ describe("MinterHolderV1", async function () {
...
+        .connect(this.accounts.deployer)
+        .updateArtblocksPrimarySalesPercentage(0);
+      // update artist primary split to zero
+      const proposedAddressesAndSplits = [
+        this.projectZero,
+        this.accounts.artist.address,
+        this.accounts.additional.address,
+        100,
+        this.accounts.additional2.address,
+        50,
```
Comment (jakerockland) [inline test/minter-suite-minters/MinterHolder/MinterHolderV1.test.ts:282]: Nit: these two constant values could be helpful to add smol dev comments for.

File: test/minter-suite-minters/MinterHolder/MinterHolderV1.test.ts [hunk 7]
```diff
@@ -185,6 +185,133 @@ describe("MinterHolderV1", async function () {
...
+
+    it("requires successful payment to artist additional payee", async function () {
+      // update artist additional payee to a contract that reverts on receive
+      const proposedAddressesAndSplits = [
+        this.projectZero,
+        this.accounts.artist.address,
+        this.deadReceiver.address,
+        50,
+        this.accounts.additional2.address,
+        50,
```
Comment (jakerockland) [inline test/minter-suite-minters/MinterHolder/MinterHolderV1.test.ts:246]: Nit: these two constant values could be helpful to add smol dev comments for.

File: test/minter-suite-minters/DA/MinterDAV2.common.ts [hunk 8]
```diff
@@ -27,6 +27,122 @@ export const MinterDAV2_Common = async () => {
...
+          .connect(this.accounts.deployer)
+          .updateArtblocksPrimarySalesPercentage(0);
+        // update artist primary split to zero
+        const proposedAddressesAndSplits = [
+          this.projectZero,
+          this.accounts.artist.address,
+          this.accounts.additional.address,
+          100,
+          this.accounts.additional2.address,
+          50,
```
Comment (jakerockland) [inline test/minter-suite-minters/DA/MinterDAV2.common.ts:125]: Nit: these two constant values could be helpful to add smol dev comments for.

File: test/minter-suite-minters/DA/MinterDAV2.common.ts [hunk 9]
```diff
@@ -27,6 +27,122 @@ export const MinterDAV2_Common = async () => {
...
+
+      it("requires successful payment to artist additional payee", async function () {
+        // update artist additional payee to a contract that reverts on receive
+        const proposedAddressesAndSplits = [
+          this.projectZero,
+          this.accounts.artist.address,
+          this.deadReceiver.address,
+          50,
+          this.accounts.additional2.address,
+          50,
```
Comment (jakerockland) [inline test/minter-suite-minters/DA/MinterDAV2.common.ts:89]: Nit: these two constant values could be helpful to add smol dev comments for.

**Review by jakerockland — APPROVED** [review]
🔥 🔥 🔥 🔥 🔥 🔥 🔥

File: test/minter-suite-minters/SetPrice/MinterSetPriceV2.common.ts (hunk 2 above)
Comment (ryley-o) [reply test/minter-suite-minters/SetPrice/MinterSetPriceV2.common.ts]: updated 🙏

File: test/minter-suite-minters/SetPrice/MinterSetPriceERC20/MinterSetPriceERC20V2.test.ts (hunk 3 above)
Comment (ryley-o) [reply test/minter-suite-minters/SetPrice/MinterSetPriceERC20/MinterSetPriceERC20V2.test.ts]: updated 🙏

File: test/minter-suite-minters/MinterMerkle/MinterMerkleV1.test.ts (hunk 4 above)
Comment (ryley-o) [reply test/minter-suite-minters/MinterMerkle/MinterMerkleV1.test.ts]: updated 🙏

File: test/minter-suite-minters/MinterMerkle/MinterMerkleV1.test.ts (hunk 5 above)
Comment (ryley-o) [reply test/minter-suite-minters/MinterMerkle/MinterMerkleV1.test.ts]: updated 🙏

File: test/minter-suite-minters/MinterHolder/MinterHolderV1.test.ts (hunk 6 above)
Comment (ryley-o) [reply test/minter-suite-minters/MinterHolder/MinterHolderV1.test.ts]: updated 🙏

File: test/minter-suite-minters/MinterHolder/MinterHolderV1.test.ts (hunk 7 above)
Comment (ryley-o) [reply test/minter-suite-minters/MinterHolder/MinterHolderV1.test.ts]: updated 🙏

File: test/minter-suite-minters/DA/MinterDAV2.common.ts (hunk 8 above)
Comment (ryley-o) [reply test/minter-suite-minters/DA/MinterDAV2.common.ts]: updated 🙏

File: test/minter-suite-minters/DA/MinterDAV2.common.ts (hunk 9 above)
Comment (ryley-o) [reply test/minter-suite-minters/DA/MinterDAV2.common.ts]: updated 🙏

File: test/minter-suite-minters/SetPrice/MinterSetPriceV2.common.ts (hunk 1 above)
Comment (ryley-o) [reply test/minter-suite-minters/SetPrice/MinterSetPriceV2.common.ts]: updated 🙏

**Pull request merged**